### ISSUE TITLE
Unnecessary warning regarding serviceAccount deprecated alias

### ISSUE THREAD

**Issue by philipsparrow** [Shopify/kubeaudit#270]
##### ISSUE TYPE
 - [X] Bug Report
 - [ ] Feature Idea

# BUG REPORT

##### SUMMARY
Running `kubeaudit --namespace default --allPods all` on a cluster reports serviceAccount deprecation warning when both serviceAccount and serviceAccountName are both present.

##### ENVIRONMENT
* Kubeaudit version:
BuildDate="2019-10-02T21:18:00Z" Commit=c74f700c9fee72a171505d4779e12a77ef445733 Version=0.7.0
* Kubeaudit install method: Binary (I think)
* kubectl version: v1.15.9
* Kubernetes (GKE) version: v1.15.11-gke.9

##### STEPS TO REPRODUCE
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: api-service
spec:
  replicas: 1
  selector:
    matchLabels:
      app: api-service
  template:
    metadata:
      labels:
        app: api-service
        tier: frontend
    spec:
      serviceAccountName: api-service
....
```
`kubeaudit --namespace default --allPods all`

##### EXPECTED RESULTS
Expected no warnings about

> serviceAccount is a deprecated alias for ServiceAccountName, use that one instead

##### ACTUAL RESULTS

`time="2020-04-29T18:56:51-04:00" level=warning msg="serviceAccount is a deprecated alias for ServiceAccountName, use that one instead" DSA=api-service KubeType=deployment Name=api-service Namespace=default SA=api-service`

##### ADDITIONAL INFORMATION

`kubectl get deployment api-service -o yaml`
Will output both the deprecated and new field:
```
      serviceAccount: api-service
      serviceAccountName: api-service
```
This is a neat check in `kubeaudit` however perhaps it should only warn about deprecated use of `serviceAccountName` if `serviceAccount` is not set.

**Comment by jonpulsifer** [comment]
+1, I did not want to make an issue about this but generally agree we can do better with the deprecation warnings. It's more of a conformance audit than a security one, I suppose.

**Comment by genevieveluyt** [comment]
This should be fixed by #243. Please re-open if the problem remains
